R/QFeatures-missing-data.R

```diff
@@ -27,9 +32,15 @@
 ## SummarizedExperiment object
 .nNAByMargin <- function(object, MARGIN = 1) {
     x <- assay(object)
-    nNA <- apply(is.na(x), MARGIN, sum)
-    n <- ifelse(MARGIN == 1, ncol(x), nrow(x))
-    pNA <- nNA / n
+    if (nrow(x) == 0 || ncol(x) == 0) {
+        nNA <- NA_integer_
+        pNA <- NA_real_
+        names(pNA) <- ""
```

Comment:
Is this line needed? Line 46 seems to unname the vector.

Reply:
pNA is expectant to have a name (see lin 44): feature name of sample name, depending. However, in this case, when there's no assay, it doesn't, hence setting it as "". 

On line 46, on the other hand, we want the value without its name.

Reply:
My bad, I overlooked it :+1:

Reply:
Quick question, why returning a DataFrame with NAs rather than with an empty row, hence  replacing `nNA <- NA_integer` with ` nNA <- integer(0)`?